### DISCUSSION TITLE
Aamzon Forwarded Email Address

### DISCUSSION THREAD

**Discussion by 0fxyzSa0vj** [moralmunky/Home-Assistant-Mail-And-Packages#567] [Q&A] [answered]
Hello,

Having issues with the amazon forwarded email address field.  My email address is in the format abc.def.ghi@gmail.com.  If I enter this into the forwarded email field the intergration seems to fail.  If I remove the periods '.', which google doesn't really care about anyway, and enter abcdefghi@gmail.com the intergration loads fine.  However, when forwarding an email it will show as having come from abc.def.ghi@gmail.com still so doesn't really fix the issue.  Are there any work arounds without having to change my amazon account email?

![image](https://user-images.githubusercontent.com/94152294/141846836-f833e3c5-4751-42fd-a081-71e156cb2546.png)
![image](https://user-images.githubusercontent.com/94152294/141846652-f1383673-171e-4312-a72c-5bbcb8e4417b.png)

**Comment by firstof9** [comment]
What's the error it throws?

**Reply by 0fxyzSa0vj** [reply]
It looks like the first image "retrying setup: none".

**Reply by firstof9** [reply]
Need to know the error from the `home-assistant.log` file.

**Reply by 0fxyzSa0vj** [reply]
ah ok, sorry.  I'll try to grab that

**Comment by 0fxyzSa0vj** [comment]
The only errors I'm seeing are these:

2021-11-15 15:48:02 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.gmail.com) data: 'NoneType' object has no attribute 'group'
2021-11-15 15:48:02 DEBUG (MainThread) [custom_components.mail_and_packages] Finished fetching Mail and Packages (imap.gmail.com) data in 6.560 seconds (success: False)
2021-11-15 15:48:02 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: 'NoneType' object has no attribute 'group'

[log.txt](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/files/7541644/log.txt)

**Reply by 0fxyzSa0vj** [reply]
removing the periods from the 'amazon forwarded address' cleans those errors up and also shows a lot more activity in the log file, looking for fedex, ups etc that I did not see originally.

I did have an amazon delivery today (forwarded to the email I'm using on HA) that didn't show up in the intergration.

**Reply by firstof9** [reply]
Ah the log file helps a lot, what version of the integration are you running?

**Reply by 0fxyzSa0vj** [reply]
I'm on version 0.3.3-2

**Comment by firstof9** [comment, marked as answer]
This should be sorted in the next beta (beta 6).

To enable betas [follow the information on the wiki here](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/wiki/Configuration-and-Email-Settings#enabling-betas).

**Reply by 0fxyzSa0vj** [reply]
awesome, thanks for the help. will test it out when available.

**Comment by Normanras** [comment]
I know its much after November 2021, but I am on `0.3.4-b24` right now and getting the same errors as above. I just installed this today, fwiw. I was going to submit a pull request for adding iCloud emails to the integration, but it gave me a bit of trouble to set up. Once I figured it out, it threw the above error. In the integration card in `Settings > Integrations > Mail & Packages`, it shows this error: 

`User input malformed: expected str for dictionary value @ data['amazon_fwds']`

In my log, it shows the following:

`2022-05-16 16:08:38 ERROR (MainThread) [custom_components.mail_and_packages] Problem updating sensors: 'int' object has no attribute 'decode'
2022-05-16 16:08:38 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.mail.me.com) data: 'int' object has no attribute 'decode'
2022-05-16 16:08:38 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: 'int' object has no attribute 'decode'`

Am I not understanding the `Amazon Forwarded Address` section?

**Reply by firstof9** [reply]
Enter `""` in this field, unless you are forwarding emails from another email address to the one you have setup in `mail and packages`.
As far as I am aware apple's email servers won't work.

**Reply by Normanras** [reply]
Okay, got it. No wonder it was giving me trouble to even set up step 1. I guess I got lucky when it went through. Maybe I'll switch to my domain on mxroute and just forward from my existing email. Thanks for the reply.

**Comment by shedman214** [comment]
I'm getting this error. Is my forward Amazon address not right, and where would I find the forward address anyway? 
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/32603273/8865ac7e-f26d-4e45-ac0f-8a237ad65fee)

**Reply by firstof9** [reply]
Remove the `()`s 
If that is your email address enter `""` only do **NOT** enter your email that you are already checking there.

**Reply by shedman214** [reply]
my fwd email is the same on both amazon and gmail. What should i enter?

**Reply by firstof9** [reply]
Enter `""`

**Reply by shedman214** [reply]
entered that and still getting an error
Logs: 
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/32603273/76d5c352-1687-47d2-946e-c57b61e95725)
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/32603273/b5b7cbd3-2b31-4383-8fb1-9928090f3683)
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/32603273/553eb567-1c93-4f17-8e0e-d7d7ac4b0fbe)
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/32603273/a236812a-4ac7-4f59-996c-963fb916f7eb)

**Reply by firstof9** [reply]
Get real logs from `home-assistant.log` please.
That last one indicates your username/password isn't correct.

**Reply by shedman214** [reply]
File was too large so i copied most recent: https://logpaste.com/58b4BojH

**Reply by firstof9** [reply]
Please provide your integration diagnostics. Here's an image of how to get the file
![image](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/assets/1105672/9ef191ae-bc58-4716-8396-3a284f1e52a5)

**Reply by shedman214** [reply]
[config_entry-mail_and_packages-db987e76ab740ede9fc13c4d17c6c094.json](https://github.com/moralmunky/Home-Assistant-Mail-And-Packages/files/14274310/config_entry-mail_and_packages-db987e76ab740ede9fc13c4d17c6c094.json)

**Reply by firstof9** [reply]
Can you see if it'll let you leave the amazon fwd section blank?
Blank or `(none)` try both please.

**Reply by shedman214** [reply]
Tried both and no luck. They did not work

**Reply by firstof9** [reply]
When you leave it blank does it give you an error?

**Reply by shedman214** [reply]
I redownloaded the mail and packages and restarted HA. added (none) so far its working again

**Reply by firstof9** [reply]
Weird, a wins a win I guess
